artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -2337,16 +2337,23 @@ public void destroyQueue(final SimpleString queueName,
                             final SecurityAuth session,
                             final boolean checkConsumerCount,
                             final boolean removeConsumers,
-                            final boolean forceAutoDeleteAddress) throws Exception {
-      destroyQueue(queueName, session, checkConsumerCount, removeConsumers, forceAutoDeleteAddress, false);
+                            final boolean instantAutoDeleteAddress) throws Exception {
+      destroyQueue(queueName, session, checkConsumerCount, removeConsumers, instantAutoDeleteAddress, false);
    }
 
+   /**
+    * instantAutoDeleteAddress will immediately remove any auto-created address when the last queue for that address is removed.
+    * Normally the reaper should be responsible for dealing with that but if this parameter is set it will be removed instantly.
```

Comment:
As earlier comment, somewhat overstates the immediacy. It wont be removed if there is a auto delete delay configured. It will then never be removed if the reaper isnt active for the address (as in case that caused the report).

This should really be javadoc rather than just a comment, and be on the interface rather than the impl.